Fuzzy alternative. (#99)
Igalia/explainers: spell-check-dictionary/README.md

Reviewer: ```suggestion
* Provide a solution which, while not specifically about autocomplete, autocorrect or spelling suggestions, also can potentially inform and or integrate with other proposals. That is, to offer something that is not in conflict with those as possible future goals.
```

Reviewer: ```suggestion
* This API does not itself aim to address spell-check suggestions, autocorrect, or AI proofreading features.
```

<details>
Choosing scope for the API largely depends on use cases. For example, Document-scoped is enough for whole-page vocabularies while subtree-scoped wins when multiple forms on one page need distinct vocabularies, or when web components want isolation. We would say that  Document scope and DOM-subtree scope are not mutually exclusive — they compose additively.

We propose to proceed with document-scoped first as it's a more conservative, easier-to-spec choice but leaves the door open for adding a partial interface HTMLElement later if multi-vocabulary scenarios emerge. 
</details>

### 6. Fuzzy / pattern matching for entries

<details>

This proposal treats each entry as a literal word rather than a structured format carrying morphological information, matching the behavior of the existing browser custom dictionary. A natural question raised in [#94](https://github.com/Igalia/explainers/issues/94) is whether to allow some regexp-like syntax in this list, so that morphological variants collapse into a single entry rather than being enumerated one inflection at a time. For example:

```
"Nutrimatic('s)?",
"Improbab(ility|le)",
"jettison(ed)?",
"babel(fish)?"
```

> and with an optional `:i` sigil for case-insensitivity. The motivation is compelling — enumerating every inflection could be a real authoring burden. However, we propose to defer this to a later iteration and revisit patterns as a backward-compatible extension. If a fuzzy syntax or pattern matching is worth having, we think the natural next step would be specifying a strict mini-grammar — a closed set of special operators, with everything else treated as literal.

Reviewer: ```suggestion
perhap with an optional `:i` sigil for case-insensitivity. The motivation is compelling — enumerating every inflection could be a real authoring burden.

We propose to defer this to a later iteration and revisit patterns as it seems like a backward-compatible extension: the tokenizers treat those most necessary characters as word boundaries. If a fuzzy syntax or pattern matching is worth having, we think the natural next step would be specifying a strict mini-grammar — a closed set of special operators, with everything else treated as literal.

Another reason to consider this as a follow on step is that it is unclear if and how these lists can and should be used by things like autocomplete or autosuggest, so we would like to leave this as simple as possible in the first iteration.
```
